eZ/Publish/Core/Repository/ContentService.php

```diff
@@ -943,6 +943,11 @@ protected function publishUrlAliasesForContent(APIContent $content, $updatePathI
                     $updatePathIdentificationString ? $languageCode === $content->contentInfo->mainLanguageCode : false
                 );
             }
+            $this->persistenceHandler->urlAliasHandler()->locationContentPublished(
```

Comment:
wouldn't it be possible in code above to detect a language code not part of content? ref `foreach ($urlAliasNames as $languageCode => $name) {`

If so we could maybe use that info to use existing method or? Cause the relationship between `publishUrlAliasForLocation` and `locationContentPublished` seems a bit strange.

Reply:
I'd need to fetch via urlAliasHandler() current language masks (possibly multiple records) as this information is no longer available on Content at this point. I'm also not sure if `publishUrlAliasForLocation` can handle removing id from mask as it performs bitwise OR (maybe cleanup ocurring later on fixes that).

I'll look into that once again, maybe I missed some obvious possibility.

Reply:
> I'm also not sure if publishUrlAliasForLocation can handle removing id from mask as it performs bitwise OR (maybe cleanup ocurring later on fixes that).

hmm, I was not thinking `publishUrlAliasForLocation` should remove, it seems to clearly only deal with one language code at a time.

Reply:
Meaning something in the direction of 8ec36f3?
If so, then I'd move entire logic to `UrlAlias\Handler` as location of language codes inside UrlAlias structure is bizarre.